wfe2/verify.go

```diff
@@ -18,233 +20,501 @@ import (
 	"gopkg.in/square/go-jose.v2"
 )
 
-// signatureValidationError indicates that the user's signature could not
-// be verified, either through adversarial activity, or misconfiguration of
-// the user client.
-type signatureValidationError string
-
-func (e signatureValidationError) Error() string { return string(e) }
-
-func algorithmForKey(key *jose.JSONWebKey) (string, error) {
+func sigAlgorithmForKey(key *jose.JSONWebKey) (jose.SignatureAlgorithm, error) {
 	switch k := key.Key.(type) {
-	case *rsa.PublicKey:
-		return string(jose.RS256), nil
+	case *rsa.PublicKey, rsa.PublicKey:
```

Comment:
I'm probably missing something but is there a reason you add the non-pointer `rsa.PublicKey` but not `ecdsa.PublicKey`?